src/mlpack/bindings/go/CMakeLists.txt

```diff
@@ -24,13 +24,15 @@ if (BUILD_GO_BINDINGS)
     find_package(Gonum)
     if (NOT GO_FOUND OR NOT GONUM_FOUND)
       unset(BUILD_GO_BINDINGS CACHE)
+      set(BUILD_GO_SHLIB OFF)
```

Comment:
I imagine that this shouldn't be needed because of the `FATAL_ERROR` that comes next?

Reply:
No problem I can remove this, but there is no harm too. 
As when user use `-DBUILD_GO_BINDINGS=ON` and if it cause `FATAL_ERROR`, Again if user use `-DBUILD_GO_BINDINGS=OFF` without specifying `-DBUILD_GO_SHLIB=ON` but still we are building all the go shared libraries because `-DBUILD_GO_SHLIB=ON` internally. But it doesn't seem a practical situation, your call I can remove this.

Reply:
I think it's fine as is, no worries. :+1: